doc/pages/modes.asciidoc

```diff
@@ -0,0 +1,102 @@
+= Modes
+
+== Description
+
+Kakoune is a modal editor which means that keys have different effects depending
+on the current mode. Therefore, modes can be conceptualized as a way to group
+related behaviors together during a text editing workflow.
+
+By default Kakoune starts in Normal mode. A few keys let users enter other
+modes where they can focus on a specific task before going back to Normal mode
+explicitely with `<esc>` or automatically in the case of *one shot* modes.
+
+Modes are stored in a stack with the top of the stack being the active mode.
+So in some scenarios, the Normal mode may feel *nested* in another one.
+The `ModeChange` hook is triggered each time a mode is popped or pushed
+on this stack. See <<hooks#,`:doc hooks`>>
+
+To get a comprehensive list of commands available for each modes,
+see <<keys#,`:doc keys`>>.
+Most of them are recap in *info* boxes in real-time if the `autoinfo` option is set.
+
+To customize key mappings in various modes, refer to <<mapping#,`:doc mapping`>>.
+
+== Builtin modes
+
+=== Normal mode
+
+Normal mode is the starting mode mainly dedicated to move, extend or create
+selections where users spend most of their time (besides inserting content).
+It also serves as a *basecamp* to access other modes.
+
+See normal commands <<keys#movement,`:doc keys movement`>>.
+
+=== Insert mode
+
+In Insert mode, typing keys add content to the current buffer like in many other
+text editors. There are several ways to reach Insert Mode and start editing at
+common locations for maximum efficiency (after cursor, at the beginning of the line…).
+See changes <<keys#changes,`:doc keys changes`>>.
+
+Use `<esc>` to go back to Normal Mode.
+Use `<a-;>` to temporary go back to Normal mode for one command.
```

Comment:
It might be worth noting that if the command you execute in "temporary normal mode" is another "enter insert mode" command, you'll wind up with two insert modes stacked together, and have to hit Escape twice to get back to normal mode. That may sound like an odd thing to do, but every few weeks somebody files a "sometimes I need to hit Escape twice" bug.